ts/query.go

```diff
@@ -406,22 +405,23 @@ func (db *DB) Query(query proto.TimeSeriesQueryRequest_Query, r Resolution,
 	// query.
 	startKey := MakeDataKey(query.Name, "" /* source */, r, startNanos)
 	endKey := MakeDataKey(query.Name, "" /* source */, r, endNanos).PrefixEnd()
-	scan := client.Scan(startKey, endKey, 0)
-	if err := db.kv.Run(scan); err != nil {
+	sr, err := db.db.Scan(startKey, endKey, 0)
+	if err != nil {
 		return nil, nil, err
 	}
-	scanResponse := scan.Reply.(*proto.ScanResponse)
 
 	// Construct a new dataSpan for each distinct source encountered in the
 	// query. Each dataspan will contain all data queried from the same source.
 	sourceSpans := make(map[string]*dataSpan)
-	for _, row := range scanResponse.Rows {
-		_, source, _, _ := DecodeDataKey(row.Key)
-		data, err := proto.InternalTimeSeriesDataFromValue(&row.Value)
-		if err != nil {
+	for _, row := range sr.Rows {
+		// TODO(pmattis): We're not checking that value tag is _CR_TS. There isn't
+		// a good way to get that info currently.
```

Comment:
I don't think we need a TODO here.

Verifying the tag isn't important when reading a known time series value, it's mostly important when writing; the C++ code needs the _CR_TS tag to identify time series data in the merge operator.

Checking for the tag while reading would only be important if the data *might* be a time series; however, in this context the only valid values are time series, so the unmarshal call is enough.

Originally I was checking for the tag on reads just to be thorough, but I don't think it was ever particularly necessary.